cdc/changefeed.go

```diff
@@ -706,42 +726,34 @@ func (c *changeFeed) handleDDL(ctx context.Context) error {
 		c.ddlJobHistory = c.ddlJobHistory[1:]
 		c.ddlExecutedTs = todoDDLJob.BinlogInfo.FinishedTS
 		c.ddlState = model.ChangeFeedSyncDML
+		c.ddlEventCache = nil
 		return nil
 	}
 
-	executed := false
 	if !c.cyclicEnabled || c.info.Config.Cyclic.SyncDDL {
 		failpoint.Inject("InjectChangefeedDDLError", func() {
 			failpoint.Return(cerror.ErrExecDDLFailed.GenWithStackByArgs())
 		})
 
 		ddlEvent.Query = binloginfo.AddSpecialComment(ddlEvent.Query)
 		log.Debug("DDL processed to make special features mysql-compatible", zap.String("query", ddlEvent.Query))
-		err = c.sink.EmitDDLEvent(ctx, ddlEvent)
+
+		c.ddlEventCache = ddlEvent
+		cdcCtx := cdcContext.NewContext(ctx, &cdcContext.GlobalVars{})
+		log.Info("sent ddl to asyncSink", zap.Reflect("ddlEvent", c.ddlEventCache))
+		done, err := c.sink.EmitDDLEvent(cdcCtx, c.ddlEventCache)
 		// If DDL executing failed, pause the changefeed and print log, rather
 		// than return an error and break the running of this owner.
 		if err != nil {
-			if cerror.ErrDDLEventIgnored.NotEqual(err) {
-				c.ddlState = model.ChangeFeedDDLExecuteFailed
-				log.Error("Execute DDL failed",
-					zap.String("ChangeFeedID", c.id),
-					zap.Error(err),
-					zap.Reflect("ddlJob", todoDDLJob))
-				return cerror.ErrExecDDLFailed.GenWithStackByArgs()
-			}
-		} else {
-			executed = true
+			return cerror.ErrExecDDLFailed.GenWithStackByArgs()
+		}
+		if done {
+			c.ddlJobHistory = c.ddlJobHistory[1:]
+			c.ddlExecutedTs = todoDDLJob.BinlogInfo.FinishedTS
+			c.ddlState = model.ChangeFeedSyncDML
+			c.ddlEventCache = nil
 		}
 	}
-	if executed {
-		log.Info("Execute DDL succeeded", zap.String("changefeed", c.id), zap.Reflect("ddlJob", todoDDLJob))
-	} else {
-		log.Info("Execute DDL ignored", zap.String("changefeed", c.id), zap.Reflect("ddlJob", todoDDLJob))
```

Comment:
We need to preserve this log

Reply:
ok

Reply:
Thes log will print  in `async_sink`.

Reply:
Please make sure that these logs will be printed exactly the same way, or otherwise `ddl_reentrant` test will not pass.

Reply:
thank you very much~!